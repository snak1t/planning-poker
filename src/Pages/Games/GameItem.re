open Ant;

let component = ReasonReact.statelessComponent("GameItem");

let make =
    (
      ~title: string,
      ~description: string,
      ~id: string,
      ~onNavigateToTask: string => unit,
      ~onDeleteGame: string => unit,
      _children,
    ) => {
  let onPlay = _e => onNavigateToTask(id);
  let onDelete = _e => onDeleteGame(id);
  let renderDescription = (desc: string) =>
    ReasonReact.string(
      switch (desc) {
      | "" => "There is no description"
      | _ => desc
      },
    );
  {
    ...component,
    render: _self =>
      <Card
        style={ReactDOMRe.Style.make(~margin="0.4rem", ())}
        title
        actions=[|
          <Icon _type=Icon.PlayCircle onClick=onPlay />,
          <Icon _type=Icon.Delete onClick=onDelete />,
        |]>
        {renderDescription(description)}
      </Card>,
  };
};

[@bs.deriving abstract]
type jsProps = {
  title: string,
  description: string,
  onNavigateToTask: string => unit,
  onDeleteGame: string => unit,
  _id: string,
};

let default =
  ReasonReact.wrapReasonForJs(~component, jsProps =>
    make(
      ~title=jsProps##title,
      ~description=jsProps##description,
      ~id=jsProps##_id,
      ~onNavigateToTask=jsProps##onNavigateToTask,
      ~onDeleteGame=jsProps##onDeleteGame,
      [||],
    )
  );