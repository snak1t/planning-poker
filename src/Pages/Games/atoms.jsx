import styled from 'styled-components';

export const GamesHeader = styled.header`
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: 0.4rem 1rem;
    h1 {
        font-size: 1.2rem;
        line-height: 1.4rem;
    }
`;

export const GamesGrid = styled.section`
    padding: 2rem;
    display: grid;
    grid-template-columns: repeat(5, 1fr);
    grid-gap: 1rem;
    min-height: 0;
    min-width: 0;
    @media screen and (max-width: 400px) {
        padding: 0.2rem;
        grid-template-columns: 1fr;
    }
`;

export const GameGridItem = styled.article`
    min-width: 0;
`;
