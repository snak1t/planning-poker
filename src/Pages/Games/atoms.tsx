import styled from 'styled-components';

export const GamesHeader = styled.header`
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: 0.4rem 1rem;
    h1 {
        font-size: 1.2rem;
        line-height: 1.4rem;
    }
`;

export const GamesWrapper = styled.section`
    padding: 2rem;
`;
