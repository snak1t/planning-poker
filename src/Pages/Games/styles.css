.Games-container {
    flex: 1;
}

.Games-list {
    display: flex;
    flex-wrap: wrap;
}

.Games-item {
    flex-basis: 300px;
    margin: 10px;
    min-height: 200px;
    display: flex;
    flex-direction: column;
    box-sizing: border-box;
    background-color: #fff;
    padding: 10px;
    box-shadow: 0 0 6px 1px rgba(0, 0, 0, 0.38);
}

.Games-gameHeader {
    line-height: 1.6rem;
    font-size: 1.2rem;
    text-align: center;
    font-weight: bold;
}

.Games-gameDescription {
    flex-grow: 1;
    display: flex;
    align-items: center;
    justify-content: center;
    text-align: center;
}

.Games-gameFooter {
    margin: 10px 0px;
    display: flex;
    justify-content: space-around;
}
