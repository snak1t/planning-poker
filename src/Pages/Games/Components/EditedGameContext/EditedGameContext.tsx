import React, { useState, useEffect } from 'react';
import { Game } from '../../../../Data/Games/GamesContext';

type EditedGameContextType = {
    game: Game | null;
    updateField<T extends keyof Game>(field: T, value: Game[T]): void;
    submitGame: () => void;
};

type EditedGameProps = {
    game: Game | null;
    onSubmit: (game: Game) => void;
};

export const EditedGameContext = React.createContext<EditedGameContextType>({
    game: null,
    updateField: () => {},
    submitGame: () => {},
});

export const EditedGameProvider: React.SFC<EditedGameProps> = ({ game, onSubmit, children }) => {
    const [clonedGame, updateClonedGame] = useState<Game | null>(game);
    useEffect(
        () => {
            updateClonedGame(game);
        },
        [game],
    );
    const updateField: EditedGameContextType['updateField'] = (key, value) => {
        return updateClonedGame(prevGame => (prevGame ? { ...prevGame, [key]: value } : null));
    };

    const submitGame: EditedGameContextType['submitGame'] = () => {
        if (clonedGame) {
            onSubmit(clonedGame);
        }
    };

    return (
        <EditedGameContext.Provider value={{ game: clonedGame, updateField, submitGame }}>
            {children}
        </EditedGameContext.Provider>
    );
};
