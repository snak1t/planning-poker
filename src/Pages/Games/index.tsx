import React, { useState } from 'react';
import { not } from 'ramda';
import { Button } from 'antd';

import { GamesContainer } from './Components/GamesContainer/GamesContainer';
import { GameAddForm } from './Components/GameAddForm/GameAddForm';
import * as Atoms from './atoms';
import { QuickGame } from '../../components/QuickGame';
import { RouteComponentProps } from 'react-router';
import { FloatButtonsContainer } from '../../components/FloatButtonsContainer/FloatButtonsContainer';

const GamesPage: React.SFC<RouteComponentProps> = () => {
    const [modalWindow, setModalWindowVisibility] = useState(false);
    const toggleModalWindow = () => setModalWindowVisibility(not);

    return (
        <>
            {modalWindow ? <GameAddForm onClose={toggleModalWindow} /> : null}
            <Atoms.GamesHeader>
                <h2>Your games</h2>
                <QuickGame />
            </Atoms.GamesHeader>
            <GamesContainer />
            <FloatButtonsContainer>
                <Button onClick={toggleModalWindow} type="primary" shape="circle" icon="plus" size="large" />
            </FloatButtonsContainer>
        </>
    );
};

export default GamesPage;
