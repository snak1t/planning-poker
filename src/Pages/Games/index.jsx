import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import evolve from 'ramda/src/evolve'
import not from 'ramda/src/not'

import './styles.css'
import GamesList from './Components/List'
import GameForm from './Components/Form'

export class GamesConatainer extends React.Component {
  state = {
    modal: false
  }

  navigateToGame = id =>
    this.props.history.replace(`/game/${this.props.userLogin}/${id}`)

  toggleModalWindow = () => this.setState(evolve({ modal: not }))

  render() {
    return (
      <div className="Games-container">
        {this.state.modal
          ? <GameForm onClose={this.toggleModalWindow} />
          : null}
        <div className="Games-title">
          <h1>Your games</h1>
        </div>
        <button onClick={this.toggleModalWindow}>Add Form</button>
        <GamesList onNavigateToTask={this.navigateToGame} />
      </div>
    )
  }
}

GamesConatainer.propTypes = {
  history: PropTypes.object.isRequired,
  userLogin: PropTypes.string
}

const mapStateToProps = state => ({
  userLogin: state.user.login
})

const mapDispatchToProps = {}

export default connect(mapStateToProps, mapDispatchToProps)(GamesConatainer)
