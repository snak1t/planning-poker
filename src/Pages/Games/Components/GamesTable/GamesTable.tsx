import React, { useContext } from 'react';
import { Game } from '../../../../Data/Games/GamesContext';
import { Table, Button, Input } from 'antd';
import { ColumnProps } from 'antd/lib/table';
import { Link } from 'react-router-dom';
import { EditedGameContext } from '../EditedGameContext/EditedGameContext';

type Props = {
  games: Game[];
  getLinkUrl: (id: string) => string;
  onRemoveGame: (id: string) => void;
  onSetGameForEdit: (id: string) => void;
  onCancelEditing: () => void;
};

export const GamesTable: React.SFC<Props> = ({
  games,
  getLinkUrl,
  onRemoveGame,
  onSetGameForEdit,
  onCancelEditing,
}) => {
  const { game: editedGame, updateField, submitGame } = useContext(
    EditedGameContext,
  );
  const columns: ColumnProps<Game>[] = [
    {
      title: 'Title',
      dataIndex: 'title',
      key: 'title',
      render: (text, gameItem) => {
        if (editedGame && gameItem.id === editedGame.id) {
          return (
            <Input
              value={editedGame.title}
              onChange={event => updateField('title', event.target.value)}
              data-testid="game-title-input"
            />
          );
        }
        return (
          <Link to={getLinkUrl(gameItem.id)}>
            <span data-testid="game-title">{text}</span>
          </Link>
        );
      },
    },
    {
      title: 'Amount of stories',
      dataIndex: 'storiesCount',
      key: 'storiesCount',
    },
    {
      title: 'Game Completed?',
      dataIndex: 'isCompleted',
      key: 'isCompleted',
      render: (text: string, gameItem: Game) => (
        <span>{gameItem.isCompleted ? 'Yes' : 'No'}</span>
      ),
    },
    {
      title: 'Actions',
      key: 'actions',
      render: (text: string, gameItem: Game) => (
        <Button.Group>
          {editedGame && editedGame.id === gameItem.id ? (
            <>
              <Button
                type="primary"
                onClick={submitGame}
                data-testid="game-submit-button"
              >
                Update game
              </Button>
              <Button
                onClick={onCancelEditing}
                data-testid="game-cancel-edit-btn"
              >
                Cancel
              </Button>
            </>
          ) : (
            <>
              {gameItem.isCompleted ? null : (
                <Button
                  onClick={() => {
                    onSetGameForEdit(gameItem.id);
                  }}
                  data-testid="game-edit-btn"
                >
                  Edit
                </Button>
              )}
              <Button
                onClick={() => {
                  onRemoveGame(gameItem.id);
                }}
                data-testid="game-remove-btn"
              >
                Remove
              </Button>
            </>
          )}
        </Button.Group>
      ),
    },
  ];
  return (
    <Table
      rowKey="id"
      defaultExpandAllRows={true}
      expandedRowRender={(record: Game) =>
        editedGame && editedGame.id === record.id ? (
          <Input
            data-testid="game-description-input"
            value={editedGame.description}
            onChange={event => updateField('description', event.target.value)}
          />
        ) : (
          <p style={{ margin: 0 }}>
            {record.description || 'No description for game provided'}
          </p>
        )
      }
      columns={columns}
      dataSource={games}
    />
  );
};
