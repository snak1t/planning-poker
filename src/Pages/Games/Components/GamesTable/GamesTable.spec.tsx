import React from 'react';
import { render, cleanup, fireEvent } from 'react-testing-library';
import { GamesTable } from './GamesTable';
import { Game } from '../../../../Data/Games/GamesContext';
import { MemoryRouter } from 'react-router';
import { EditedGameProvider } from '../EditedGameContext/EditedGameContext';

const mockedGames: Game[] = [
  {
    id: '1',
    description: 'Game1 description',
    title: 'Game1 title',
    isCompleted: false,
    storiesCount: 1,
    user: 'user1',
  },
  {
    id: '2',
    description: 'Game2 description',
    title: 'Game2 title',
    isCompleted: true,
    storiesCount: 2,
    user: 'user1',
  },
];

describe('<GamesTable:', () => {
  afterEach(cleanup);
  let getLinkUrl: (id: string) => string;
  let onRemoveGame: (id: string) => void;
  let onEditGame: (id: string) => void;
  let onCancelEditing: () => void;
  beforeEach(() => {
    getLinkUrl = jest
      .fn()
      .mockName('getLinkUrl')
      .mockReturnValue('/some-link');
    onRemoveGame = jest.fn().mockName('onRemoveGame');
    onEditGame = jest.fn().mockName('onEditGame');
    onCancelEditing = jest.fn().mockName('onCancelEditing');
  });

  it('should be rendered', () => {
    const { container } = render(
      <MemoryRouter>
        <GamesTable
          onSetGameForEdit={onEditGame}
          onCancelEditing={onCancelEditing}
          games={mockedGames}
          getLinkUrl={getLinkUrl}
          onRemoveGame={onRemoveGame}
        />
      </MemoryRouter>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render links', () => {
    render(
      <MemoryRouter>
        <GamesTable
          onSetGameForEdit={onEditGame}
          onCancelEditing={onCancelEditing}
          games={mockedGames}
          getLinkUrl={getLinkUrl}
          onRemoveGame={onRemoveGame}
        />
      </MemoryRouter>,
    );
    expect(getLinkUrl).toHaveBeenCalledTimes(2);
    expect(getLinkUrl).toHaveBeenLastCalledWith(mockedGames[1].id);
  });

  it('should render two titles in normal mode', () => {
    const { getAllByTestId } = render(
      <MemoryRouter>
        <EditedGameProvider game={null} onSubmit={() => {}}>
          <GamesTable
            onSetGameForEdit={onEditGame}
            onCancelEditing={onCancelEditing}
            games={mockedGames}
            getLinkUrl={getLinkUrl}
            onRemoveGame={onRemoveGame}
          />
        </EditedGameProvider>
      </MemoryRouter>,
    );

    const gameTitles = getAllByTestId('game-title');
    expect(gameTitles.length).toBe(2);
  });

  it('should correctly remove game on clicking', () => {
    const { getByTestId } = render(
      <MemoryRouter>
        <EditedGameProvider game={null} onSubmit={() => {}}>
          <GamesTable
            onSetGameForEdit={onEditGame}
            onCancelEditing={onCancelEditing}
            games={mockedGames.slice(0, 1)}
            getLinkUrl={getLinkUrl}
            onRemoveGame={onRemoveGame}
          />
        </EditedGameProvider>
      </MemoryRouter>,
    );

    const deleteBtn = getByTestId('game-remove-btn');

    fireEvent.click(deleteBtn);
    expect(onRemoveGame).toHaveBeenCalled();
    expect(onRemoveGame).toHaveBeenCalledWith(mockedGames[0].id);
  });

  it('should call onSetEditGame when Edit button is clicked', () => {
    const { getByTestId } = render(
      <MemoryRouter>
        <EditedGameProvider game={null} onSubmit={() => {}}>
          <GamesTable
            onSetGameForEdit={onEditGame}
            onCancelEditing={onCancelEditing}
            games={mockedGames.slice(0, 1)}
            getLinkUrl={getLinkUrl}
            onRemoveGame={onRemoveGame}
          />
        </EditedGameProvider>
      </MemoryRouter>,
    );

    const editBtn = getByTestId('game-edit-btn');

    fireEvent.click(editBtn);
    expect(onEditGame).toHaveBeenCalled();
    expect(onEditGame).toHaveBeenCalledWith(mockedGames[0].id);
  });

  it('should render input elements and submit button for edited field', () => {
    const { getByTestId } = render(
      <MemoryRouter>
        <EditedGameProvider game={mockedGames[0]} onSubmit={() => {}}>
          <GamesTable
            onSetGameForEdit={onEditGame}
            onCancelEditing={onCancelEditing}
            games={mockedGames.slice(0, 1)}
            getLinkUrl={getLinkUrl}
            onRemoveGame={onRemoveGame}
          />
        </EditedGameProvider>
      </MemoryRouter>,
    );
    const titleInput = getByTestId('game-title-input') as HTMLInputElement;
    expect(titleInput).toBeDefined();
    expect(titleInput.value).toBe(mockedGames[0].title);

    const descriptionInput = getByTestId(
      'game-description-input',
    ) as HTMLInputElement;
    expect(descriptionInput).toBeDefined();
    expect(descriptionInput.value).toBe(mockedGames[0].description);

    expect(getByTestId('game-submit-button')).toBeDefined();
  });

  it('should render one title if one of the game is in edit mode', () => {
    const { getAllByTestId } = render(
      <MemoryRouter>
        <EditedGameProvider game={mockedGames[0]} onSubmit={() => {}}>
          <GamesTable
            onSetGameForEdit={onEditGame}
            onCancelEditing={onCancelEditing}
            games={mockedGames}
            getLinkUrl={getLinkUrl}
            onRemoveGame={onRemoveGame}
          />
        </EditedGameProvider>
      </MemoryRouter>,
    );

    const gameTitles = getAllByTestId('game-title');
    expect(gameTitles.length).toBe(1);
  });

  it('should not render edit button for already played game', () => {
    const { queryByText } = render(
      <MemoryRouter>
        <EditedGameProvider game={null} onSubmit={() => {}}>
          <GamesTable
            onSetGameForEdit={onEditGame}
            onCancelEditing={onCancelEditing}
            games={mockedGames.slice(-1)}
            getLinkUrl={getLinkUrl}
            onRemoveGame={onRemoveGame}
          />
        </EditedGameProvider>
      </MemoryRouter>,
    );

    expect(queryByText('Edit')).toBeNull();
  });
});
