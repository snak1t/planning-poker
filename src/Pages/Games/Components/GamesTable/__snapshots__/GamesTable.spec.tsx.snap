// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GamesTable: should be rendered 1`] = `
<div
  class="ant-table-wrapper"
>
  <div
    class="ant-spin-nested-loading"
  >
    <div
      class="ant-spin-container"
    >
      <div
        class="ant-table ant-table-default ant-table-scroll-position-left"
      >
        <div
          class="ant-table-content"
        >
          <div
            class="ant-table-body"
          >
            <table
              class=""
            >
              <colgroup>
                <col
                  class="ant-table-expand-icon-col"
                />
                <col />
                <col />
                <col />
                <col />
              </colgroup>
              <thead
                class="ant-table-thead"
              >
                <tr>
                  <th
                    class="ant-table-expand-icon-th"
                    rowspan="1"
                    title=""
                  />
                  <th
                    class=""
                  >
                    <div>
                      Title
                    </div>
                  </th>
                  <th
                    class=""
                  >
                    <div>
                      Amount of stories
                    </div>
                  </th>
                  <th
                    class=""
                  >
                    <div>
                      Game Completed?
                    </div>
                  </th>
                  <th
                    class=""
                  >
                    <div>
                      Actions
                    </div>
                  </th>
                </tr>
              </thead>
              <tbody
                class="ant-table-tbody"
              >
                <tr
                  class="ant-table-row ant-table-row-level-0"
                  data-row-key="1"
                >
                  <td
                    class="ant-table-row-expand-icon-cell"
                  >
                    <span
                      class="ant-table-row-expand-icon ant-table-row-expanded"
                    />
                  </td>
                  <td
                    class=""
                  >
                    <a
                      href="/some-link"
                    >
                      <span
                        data-testid="game-title"
                      >
                        Game1 title
                      </span>
                    </a>
                  </td>
                  <td
                    class=""
                  >
                    1
                  </td>
                  <td
                    class=""
                  >
                    <span>
                      No
                    </span>
                  </td>
                  <td
                    class=""
                  >
                    <div
                      class="ant-btn-group"
                    >
                      <button
                        class="ant-btn"
                        data-testid="game-edit-btn"
                        type="button"
                      >
                        <span>
                          Edit
                        </span>
                      </button>
                      <button
                        class="ant-btn"
                        data-testid="game-remove-btn"
                        type="button"
                      >
                        <span>
                          Remove
                        </span>
                      </button>
                    </div>
                  </td>
                </tr>
                <tr
                  class="ant-table-expanded-row ant-table-expanded-row-level-1"
                  data-row-key="1-extra-row"
                >
                  <td
                    class=""
                  />
                  <td
                    class=""
                    colspan="4"
                  >
                    <p
                      style="margin: 0px;"
                    >
                      Game1 description
                    </p>
                  </td>
                </tr>
                <tr
                  class="ant-table-row ant-table-row-level-0"
                  data-row-key="2"
                >
                  <td
                    class="ant-table-row-expand-icon-cell"
                  >
                    <span
                      class="ant-table-row-expand-icon ant-table-row-expanded"
                    />
                  </td>
                  <td
                    class=""
                  >
                    <a
                      href="/some-link"
                    >
                      <span
                        data-testid="game-title"
                      >
                        Game2 title
                      </span>
                    </a>
                  </td>
                  <td
                    class=""
                  >
                    2
                  </td>
                  <td
                    class=""
                  >
                    <span>
                      Yes
                    </span>
                  </td>
                  <td
                    class=""
                  >
                    <div
                      class="ant-btn-group"
                    >
                      <button
                        class="ant-btn"
                        data-testid="game-remove-btn"
                        type="button"
                      >
                        <span>
                          Remove
                        </span>
                      </button>
                    </div>
                  </td>
                </tr>
                <tr
                  class="ant-table-expanded-row ant-table-expanded-row-level-1"
                  data-row-key="2-extra-row"
                >
                  <td
                    class=""
                  />
                  <td
                    class=""
                    colspan="4"
                  >
                    <p
                      style="margin: 0px;"
                    >
                      Game2 description
                    </p>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
      <ul
        class="ant-pagination ant-table-pagination"
        unselectable="unselectable"
      >
        <li
          aria-disabled="true"
          class="ant-pagination-disabled ant-pagination-prev"
          title="Previous Page"
        >
          <a
            class="ant-pagination-item-link"
          >
            <i
              class="anticon anticon-left"
            >
              <svg
                aria-hidden="true"
                class=""
                data-icon="left"
                fill="currentColor"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 0 0 0 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                />
              </svg>
            </i>
          </a>
        </li>
        <li
          class="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
          tabindex="0"
          title="1"
        >
          <a>
            1
          </a>
        </li>
        <li
          aria-disabled="true"
          class="ant-pagination-disabled ant-pagination-next"
          title="Next Page"
        >
          <a
            class="ant-pagination-item-link"
          >
            <i
              class="anticon anticon-right"
            >
              <svg
                aria-hidden="true"
                class=""
                data-icon="right"
                fill="currentColor"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M765.7 486.8L314.9 134.7A7.97 7.97 0 0 0 302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 0 0 0-50.4z"
                />
              </svg>
            </i>
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
